pandas/_libs/lib.pyx

```diff
@@ -2247,6 +2258,28 @@ def maybe_convert_numeric(
     cdef:
         object val = values[0]
 
+    # Convert python strings into ones readable by C
+
+    cdef char* tsep
+    cdef char* dsep
+    # Use null char to represent lack of separator
+    if thousands is None:
+        tsep = "\0"
```

Comment:
I think this is confusing to assign the nul byte instead of just assigning to NULL, especially with how Cython handles `char *` like this (although the comment above about using char declarations in the function signature would probably simplify this)

Reply:
I did it this way as precise_xstrtod() in the C parser uses the null byte to represent the lack of a separator when processing a string (see line 1633 of tokenizer.c), previously this was just hardcoded as one of its parameters, but unless we want to change the C implementation then tsep needs to be set to the null char at some point, it could be done in C rather than cython if you think that's neater.